playbook/app/pb_kits/playbook/utilities/globalProps.ts

```diff
@@ -0,0 +1,156 @@
+type Sizes = "xs" | "sm" | "md" | "lg" | "xl"
+type None = "none"
+
+type AllSizes = None & Sizes
+
+type Margin = {
+  marginRight?: AllSizes,
+  marginLeft?: AllSizes,
+  marginTop?: AllSizes,
+  marginBottom?: AllSizes,
+  marginX?: AllSizes,
+  marginY?: AllSizes,
+  margin?: AllSizes,
+}
+
+type Padding = {
+  paddingRight?: AllSizes,
+  paddingLeft?: AllSizes,
+  paddingTop?: AllSizes,
+  paddingBottom?: AllSizes,
+  paddingX?: AllSizes,
+  paddingY?: AllSizes,
+  padding?: AllSizes,
+}
+
+type Dark = {
+  dark?: boolean,
+}
+
+type NumberSpacing = {
+  numberSpacing?: "tabular",
+}
+
+type MaxWidth = {
+  maxWidth?: Sizes,
+}
+
+type ZIndex = {
+  zIndex?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10,
+}
+
+type Shadow = {
+  shadow?: "none" | "deep" | "deeper" | "deepest",
+}
+
+type LineHeight = {
+  lineHeight?: "loosest" | "looser" | "loose" | "normal" | "tight" | "tighter" | "tightest",
+}
+
+type Display = {
+  display?: "block" | "flex" | "hidden" | "inline_block" | "inline" | "inline_flex",
+}
+
+type Cursor = {
+  cursor?: "pointer",
+}
+
+type BorderRadius = {
+  borderRadius?: "none" | "xs" | "sm" | "md" | "lg" | "xl" | "rounded",
+}
+
+export type GlobalProps = BorderRadius & Cursor & Dark & Display & LineHeight & Margin & MaxWidth & NumberSpacing & Padding & Shadow & ZIndex
+
+// Prop categories
+const PROP_CATEGORIES: {[key:string]: (props: {[key: string]: any}) => string} = {
```

Comment:
is the main advantage here that you are able to specify the datatype to avoid potential errors?

Reply:
Absolutely! Now we can provide developers an expected and auto-documented API for these global props. TS will prevent any out-of-bounds prop values as they type. 😎